src/appleseed/renderer/kernel/shading/shadingengine.cpp

```diff
@@ -105,17 +106,40 @@ void ShadingEngine::shade_hit_point(
     // Retrieve the material of the intersected surface.
     const Material* material = shading_point.get_material();
 
-    // Apply OSL transparency if needed.
-    if (material &&
-        material->get_render_data().m_shader_group &&
-        material->get_render_data().m_shader_group->has_transparency())
+    if (material != nullptr && material->get_render_data().m_shader_group)
```

Comment:
You're mixing implicit and explicit comparisons here, let's switch to all explicit:

    if (material != nullptr &&
        material->get_render_data().m_shader_group != nullptr)